src/r_defs.h

```diff
@@ -894,6 +894,10 @@ struct line_t
 	sector_t	*frontsector, *backsector;
 	int 		validcount;	// if == validcount, already checked
 	int			locknumber;	// [Dusk] lock number for special
+	bool		portal;
+	bool		portal_mirror;
+	bool		portal_passive;
+	line_t		*portal_dst;

```

Comment:
A bit nit-picky, but since this ultimately affects the software renderer I thought I would mention it anyway: it might be best to reorder these so the portal_dst pointer is first.  In general for best structure packing/alignment/cache friendliness you want the largest elements first.